newIDE/app/src/EventsSheet/index.js

```diff
@@ -1015,39 +1171,86 @@ export class EventsSheetComponentWithoutHandle extends React.Component<
   };
 
   undo = () => {
-    if (!canUndo(this.state.history)) return;
+    if (!canUndo(this.state.eventsHistory)) return;
 
     const { events, project } = this.props;
-    const newHistory = undo(this.state.history, events, project);
+    const newEventsHistory = undo(this.state.eventsHistory, events, project);
 
     // /!\ Events were changed, so any reference to an existing event can now
     // be invalid. Make sure to immediately trigger a forced update before
     // any re-render that could use a deleted/invalid event.
-    if (this._eventsTree) this._eventsTree.forceEventsUpdate();
-
-    // /!\ Also clear selection, that can contain reference to invalid events or
-    // events not shown on screen.
-    this.setState({ history: newHistory, selection: clearSelection() }, () =>
-      this.updateToolbar()
-    );
+    if (this._eventsTree)
```

Comment:
Can you do `const {_eventsTree: eventsTree} = this` so that you don't have to do checks below?